tools/c7n_gcp/c7n_gcp/resources/dataflow.py

```diff
@@ -31,3 +42,23 @@ def get(client, event):
                     'jobId': jmespath.search('protoPayload.request.job_id', event)
                 }
             )
+
+    def augment(self, resources):
```

Comment:
so typically we'd want to let users express job states as a 'query' field on the policy which is a pass through server side filter rather than client side filtering, as the cardinalities as quite differentiated typically, and then default the query to using ACTIVE, this is similiar to what we do with aws.emr jobs.  re apis per https://cloud.google.com/dataflow/docs/reference/rest/v1b3/Filter and
https://cloud.google.com/dataflow/docs/reference/rest/v1b3/projects.jobs/list

Reply:
thanks. that makes a lot of sense. will go ahead and update.

Reply:
defaults to ACTIVE now and can be overridden in query 
```
  - name: dataflow-public-ip
    resource: gcp.dataflow-job
    query: [{'filter': 'ALL'}]
```